src/File.php

```diff
@@ -1536,4 +1536,14 @@ protected function getFilter()
     {
         return FileNameFilter::create();
     }
+
+    /**
+     * Get an array with each segment (directory name and file name) in a file path as a value.
+     */
+    protected static function getFilePathParts(string $filePath): array
+    {
+        // Explicitly allow zero as a segment in the path (e.g. /some/path/0/file.txt)
+        $notEmpty = fn(mixed $part) => ($part === 0 || $part === '0' || (bool) $part);
```

Comment:
I guess the conversion is implicit here, is it not? Ok to leave as is if it's better from some perspective.

```suggestion
        $notEmpty = fn(mixed $part) => ($part === 0 || $part === '0' || $part);
```

Reply:
Yeah, I guess it is implicit. In my head it made sense to be explicit here but both do exactly the same thing. I think this is just up to subjective preference.

Reply:
I learnt not to cast to bool if it's in any logical condition/expression as PhpStorm always hints at it as useless due to it being casted implicitly.